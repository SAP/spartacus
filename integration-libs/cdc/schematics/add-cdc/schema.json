{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDCSchematics",
  "title": "CDC Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDC feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["CDC", "CDC-B2B"],
        "type": "string"
      },
      "x-prompt": {
        "message": "Which feature would you like to set up from the CDC library?",
        "type": "list",
        "items": [
          {
            "value": "CDC",
            "label": "CDC for B2C"
          },
          {
            "value": "CDC-B2B",
            "label": "CDC for B2B"
          }
        ]
      }
    },
    "javascriptUrl": {
      "type": "string",
      "description": "Location of the CDC Java Script SDK in Customer Data Cloud",
      "x-prompt": "[CDC] What is the CDC JS SDK URL?"
    },
    "baseSite": {
      "type": "string",
      "description": "Name of the Spartcus base site",
      "x-prompt": "[CDC] What is the base site?",
      "default": "electronics-spa"
    },
    "sessionExpiration": {
      "type": "number",
      "description": "Session Expiration time in seconds",
      "x-prompt": "[CDC] What is the CDC Session expiration time (in seconds)?",
      "default": 3600
    },
    "httpHeaderName": {
      "type": "string",
      "description": "Name of HTTP Header key for Reset Password",
      "x-prompt": "[CDC] What is the HTTP Header key for Reset Password?",
      "default": "apikey"
    }
  },
  "required": []
}
