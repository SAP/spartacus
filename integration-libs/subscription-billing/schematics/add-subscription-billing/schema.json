{
  "$schema": "http://json-schema.org/schema",
  "$id": "SubscriptionBillingSchematics",
  "title": "Segment Refs Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Subscription-Billing"]
    }
  }
}
