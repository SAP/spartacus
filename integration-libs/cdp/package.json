{
  "name": "@spartacus/cdp",
  "version": "2211.31.0",
  "description": "Customer Data Platform Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer data cloud",
    "gigya",
    "cdp"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cdp",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@spartacus/customer-ticketing": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
