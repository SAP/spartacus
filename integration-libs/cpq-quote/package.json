{
  "name": "@spartacus/cpq-quote",
  "version": "2211.31.0",
  "description": "cpq-quote library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "cpq-quote"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cpq-quote",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
