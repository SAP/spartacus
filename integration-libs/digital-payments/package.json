{
  "name": "@spartacus/digital-payments",
  "version": "2211.31.0",
  "description": "Digital Payments Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "digital payments",
    "payments"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/digital-payments",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ng-select/ng-select": "^12.0.4",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/checkout": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
