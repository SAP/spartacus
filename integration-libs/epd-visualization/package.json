{
  "name": "@spartacus/epd-visualization",
  "version": "2211.31.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@sapui5/ts-types-esm": "1.120.1",
    "@spartacus/cart": "2211.31.0",
    "@spartacus/core": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "@spartacus/storefront": "2211.31.0",
    "@spartacus/styles": "2211.31.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
