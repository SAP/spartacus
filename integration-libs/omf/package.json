{
  "name": "@spartacus/omf",
  "version": "2211.31.0",
  "description": "SAP Order Management Foundation Integration",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "omf"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ngrx/store": "^17.0.1",
    "@spartacus/core": "2211.31.0",
    "@spartacus/order": "2211.31.0",
    "@spartacus/schematics": "2211.31.0",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
