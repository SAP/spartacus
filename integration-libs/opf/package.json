{
  "name": "@spartacus/opf",
  "version": "2211.25.1",
  "description": "SAP Open Payment Framework integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "opf"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/opf",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/platform-browser": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@ng-select/ng-select": "^12.0.4",
    "@ngrx/store": "^17.0.1",
    "@spartacus/cart": "2211.25.1",
    "@spartacus/checkout": "2211.25.1",
    "@spartacus/core": "2211.25.1",
    "@spartacus/order": "2211.25.1",
    "@spartacus/pickup-in-store": "2211.25.1",
    "@spartacus/schematics": "2211.25.1",
    "@spartacus/storefront": "2211.25.1",
    "@spartacus/styles": "2211.25.1",
    "@spartacus/user": "2211.25.1",
    "bootstrap": "^4.6.2",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
