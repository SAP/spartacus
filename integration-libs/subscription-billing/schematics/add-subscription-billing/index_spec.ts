/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  SPARTACUS_SCHEMATICS,
  SPARTACUS_SUBSCRIPTION_BILLING,
  SUBSCRIPTION_BILLING_FEATURE_NAME,
  SpartacusOptions,
  LibraryOptions as SpartacusSubscriptionBillingOptions,
  subscriptionBillingFeatureModulePath,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus Subscription Billing Schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_SUBSCRIPTION_BILLING,
    collectionPath
  );
  let appTree: UnitTestTree;
  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };
  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };
  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };
  const libraryNoFeaturesOptions: SpartacusSubscriptionBillingOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };
  const subscriptionBillingFeatureOptions: SpartacusSubscriptionBillingOptions =
    {
      ...libraryNoFeaturesOptions,
      features: [SUBSCRIPTION_BILLING_FEATURE_NAME],
    };
  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );
    appTree = await schematicRunner.runExternalSchematic(
      SPARTACUS_SCHEMATICS,
      'ng-add',
      { ...spartacusDefaultOptions, name: 'schematics-test' },
      appTree
    );
  });
  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        libraryNoFeaturesOptions,
        appTree
      );
    });
    it('should not create any of the feature modules', () => {
      expect(appTree.exists(subscriptionBillingFeatureModulePath)).toBeFalsy();
    });
  });
  describe('Subscription Billing feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          subscriptionBillingFeatureOptions,
          appTree
        );
      });
      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });
      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(
          subscriptionBillingFeatureModulePath
        );
        expect(module).toMatchSnapshot();
      });
      it('should install the appropriate dependencies', async () => {});
    });
    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...subscriptionBillingFeatureOptions, lazy: false },
          appTree
        );
      });
      it('should import appropriate modules', async () => {
        const module = appTree.readContent(
          subscriptionBillingFeatureModulePath
        );
        expect(module).toMatchSnapshot();
      });
    });
  });
});
