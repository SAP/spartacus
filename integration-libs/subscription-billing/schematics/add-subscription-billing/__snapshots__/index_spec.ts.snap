// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Subscription Billing Schematics: ng-add Subscription Billing feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from "@spartacus/core";
import { SubscriptionBillingModule } from "@spartacus/subscription-billing";
import { subscriptionBillingTranslationChunksConfig, subscriptionBillingTranslations } from "@spartacus/subscription-billing/assets";
import { SubscriptionBillingRootModule } from "@spartacus/subscription-billing/root";

@NgModule({
  declarations: [],
  imports: [
    SubscriptionBillingRootModule,
    SubscriptionBillingModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: subscriptionBillingTranslations,
      chunks: subscriptionBillingTranslationChunksConfig,
    },
  })]
})
export class SubscriptionBillingFeatureModule { }
"
`;

exports[`Spartacus Subscription Billing Schematics: ng-add Subscription Billing feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from "@spartacus/core";
import { subscriptionBillingTranslationChunksConfig, subscriptionBillingTranslations } from "@spartacus/subscription-billing/assets";
import { SubscriptionBillingRootModule } from "@spartacus/subscription-billing/root";

@NgModule({
  declarations: [],
  imports: [
    SubscriptionBillingRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      SubscriptionBilling: {
        module: () =>
          import('@spartacus/subscription-billing').then((m) => m.SubscriptionBillingModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: subscriptionBillingTranslations,
      chunks: subscriptionBillingTranslationChunksConfig,
    },
  })
  ]
})
export class SubscriptionBillingFeatureModule { }
"
`;
